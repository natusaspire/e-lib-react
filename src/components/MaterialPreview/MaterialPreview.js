import React, { memo } from 'react';
import { Link } from 'react-router-dom';

const getMaterialLink = (endpoint, materialId) =>
  `/materials/${endpoint}/${materialId}`;

const MaterialPreview = ({ endpoint, material, image }) => (
  <div className="card">
    <div className="card__body">
      <img className="card__image" src={image} alt="" />
    </div>
    <div className="card__footer">
      <Link className="card__link" to={getMaterialLink(endpoint, material.id)}>
        {material.title}
      </Link>
    </div>
  </div>
);

export default memo(MaterialPreview);
