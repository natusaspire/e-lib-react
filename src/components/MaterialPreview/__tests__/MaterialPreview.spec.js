import React from 'react';
import { shallow } from 'enzyme';

import MaterialPreview from '@/components/MaterialPreview';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));

describe('<MaterialPreview />', () => {
  it('should render material preview', () => {
    const props = {
      endpoint: 'text',
      material: {
        id: 1,
        type: 'text',
        title: 'Material title',
        textContent: 'Material text content.'
      },
      image: 'text.png'
    };

    const wrapper = shallow(<MaterialPreview {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
