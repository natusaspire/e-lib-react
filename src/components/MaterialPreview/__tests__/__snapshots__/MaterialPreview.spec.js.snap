// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialPreview /> should render material preview 1`] = `
<div
  className="card"
>
  <div
    className="card__body"
  >
    <img
      alt=""
      className="card__image"
      src="text.png"
    />
  </div>
  <div
    className="card__footer"
  >
    <Link
      className="card__link"
      to="/materials/text/1"
    >
      Material title
    </Link>
  </div>
</div>
`;
